framework/src/org/checkerframework/common/signatureinference/SignatureInferenceScenes.java

```diff
@@ -0,0 +1,588 @@
+
+    /** Maps file paths (Strings) to Scenes. */
+    private static Map<String, AScene> scenes = new HashMap<String, AScene>();
+
+    /** Set containing all Scenes that were modified in the current ClassTree.
+     * Modifying a Scene means adding (or changing) a type annotation for a
+     * field, method return type or method parameter type in the Scene.
+     * (Scenes are modified by the method
+     * {@link SignatureInferenceScenes#updateAnnotationSetInScene updateAnnotationSetInScene)}
+     */
+    private static Set<String> modifiedScenes = new HashSet<String>();
+
+    /**
+     * Returns the Scene related to a .jaif file path passed as input.
+     */
+    private static AScene getScene(String jaifPath) {
+        if (!scenes.containsKey(jaifPath)) {
+            File jaifFile = new File(jaifPath);
+            AScene scene = new AScene();
+            if (jaifFile.exists()) {
+                try {
+                    IndexFileParser.parseFile(jaifPath, scene);
+                } catch (IOException e) {
+                    ErrorReporter.errorAbort("Could not open file in: " + jaifPath + "."
+                            + " Exception message: " + e.getMessage());
+                }
+            }
+            scenes.put(jaifPath, scene);
+        }
+        return scenes.get(jaifPath);
+    }
+
+    /**
+     * Clears the set of modified scenes.
+     * (Scenes are modified by the method
+     * {@link SignatureInferenceScenes#updateAnnotationSetInScene updateAnnotationSetInScene)}
+     */
+    public static void clearModifiedScenes() {
+        modifiedScenes.clear();
+    }
+
+    /**
+     * Adds the identifier of a Scene in the set of modified scenes.
+     * (Scenes are modified by the method
+     * {@link SignatureInferenceScenes#updateAnnotationSetInScene updateAnnotationSetInScene)}
+     */
+    private static void addModifiedScene(String scene) {
+        modifiedScenes.add(scene);
+    }
+
+    /**
+     * Write all modified scenes into .jaif files.
+     * (Scenes are modified by the method
+     * {@link SignatureInferenceScenes#updateAnnotationSetInScene updateAnnotationSetInScene)}
+     */
+    public static void writeScenesToJaif() {
+        for (String jaifPath : modifiedScenes) {
+            try {
+                File jaifFile = new File(jaifPath);
+                if (!jaifFile.exists()) {
+                    jaifFile.getParentFile().mkdirs();
+                }
+                AScene scene = scenes.get(jaifPath);
+                IndexFileWriter.write(scene, new FileWriter(jaifPath));
+            } catch (IOException e) {
+                ErrorReporter.errorAbort("Could not open file in: " + jaifPath
+                        + ". Exception message: " + e.getMessage());
+            } catch (DefException e) {
+                ErrorReporter.errorAbort(e.getMessage());
+            } catch (Exception e) {
+                System.out.println();
```

Comment:
Why are all other exceptions silently swallowed?  This doesn't even print e.  But I think it should throw an exception and halt execution (so that we developers are forced to fix it!) rather than masking the error.